### DISCUSSION TITLE
Where should i put the auto completion files for different apps ?

### DISCUSSION THREAD

**Discussion by xd003** [mattmc3/antidote#163] [Q&A] [answered]
I use many different apps that ship with their shell completions file for zsh. I just wanted to ask where exactly should i place all these files. If i am not mistaken it goes in the fpath directory which in my case is - `fpath+=(${ZDOTDIR:-~}/.antidote)`. So i should i simply drop all those shell completion files directly within the `.antidote` directory or it should go somewhere else ?

**Comment by mattmc3** [comment, marked as answer]
> I just wanted to ask where exactly should i place all these files.

Wait - don't manually move files. That's not how `fpath` works. You don't drop files in a special `fpath` location - instead, you modify the `fpath` array and add the directories you need. And antidote can do that for you with the `kind:fpath` notation.

The way Zsh's `compinit` completions work is your `fpath` array has to be fully built prior to running `compinit`. That's not an antidote thing, just how Zsh works. So, with that in mind, you want to load your plugins prior to running `compinit` where possible.

```zsh
# contents of ${ZDOTDIR:-$HOME}/.zsh_plugins.txt

# regular plugins get added to fpath automatically, in addition to being sourced
zsh-users/zsh-autosuggestions

# you can also use the fpath annotation to add locations to your fpath
zsh-users/zsh-completions kind:fpath path:src

# if you have a local folder with completions, you can add that too
/path/to/my/zsh/completions kind:fpath

# oh-my-zsh plugins get complicated because they often call functions that don't
# actually exist until after compinit is called. It's a chicken-and-egg issue. You
# need the plugin in fpath, but not sourced until after compinit.

# OMZ plugin you won't actually source until later
ohmyzsh/ohmyzsh path:plugins/rails kind:fpath

# now use a compinit plugin
belak/zsh-utils path:completion

# finally, load any plugins that required compinit to already exist
ohmyzsh/ohmyzsh path:plugins/rails
```

See:
- #74
- #144
- #127

**Reply by xd003** [reply]
Thanks for the heads up. Just as a final confirmation, say i want to include completions for [eza](https://github.com/eza-community/eza) so i put `eza-community/eza path:completions/zsh kind:fpath` before `mattmc3/zephyr path:plugins/completion` ( i am using this to call compinit ) right ?

Lastly, in the `path` option can i specify a file instead of folder ? Since some apps on github seem to have all different shell completions in a single folder itself. For example - [bat](https://github.com/sharkdp/bat/tree/master/assets/completions) , So can i do `sharkdp/bat path:assets/completions/bat.zsh.in kind:fpath` for this one ?

**Reply by lcherpit** [reply]
tx ! would be nice to add a link to this answer in https://antidote.sh/completions
